lunchbot.py

```diff
@@ -85,29 +93,26 @@ def weighted_choice(self, choices):
             upto += item['weight']
         assert False, 'Shouldn\'t get here'
 
-    def handle_lunchbot_command(self, command, user):
-        """
-        Takes a command from Slack chat directed at Lunchbot and performs the action if it is supported
-        """
-        post_ephemeral = False
-
-        if 'help' in command.lower():
-            post_ephemeral = True
-            msg = ('Here are the actions that Lunchbot can perform:\n'
-                   'list restaurants\n'
-                   'add restaurant <name> <weight>\n'
-                   'remove restaurant <name>\n')
-        elif 'list restaurants' in command.lower():
-            post_ephemeral = True
-            msg = ''
-            for restaurant in self.RESTAURANTS:
-                msg = '\n'.join([msg, '%s has weight %s' % (restaurant['name'], restaurant['weight'])])
-        elif 'add restaurant' in command.lower():
+    def add_restaurant_command(self, command):
+        try:
             # Split command but keep quoted substrings as a single unit
             pieces = [p for p in re.split("( |\\\".*?\\\"|'.*?')", command) if p.strip()]
+
             # Get restaurant and weight to add from command
             restaurant = pieces[-2]
             weight = int(pieces[-1])
+
+            # error checking.
+            if restaurant is None:
+                return "Please specify a restaurant"
+            if weight is None:
+                return "Please specify a weight"
+            else:
+                try:
+                    weight = int(weight)
+                except:
```

Comment:
We should be more specific with regards to the exceptions we are trying to catch